### ISSUE TITLE
Write tests for Wallet contract.

### ISSUE THREAD

**Issue by MicahZoltu** [paritytech/contracts#64]
https://github.com/paritytech/contracts/blob/whg/Wallet.sol

Maybe I just don't know where they live, but this contract should have a full test suite behind it, both for validation and regression.  Truffle seems to be a popular framework for contract testing, though I'm personally a fan of EthereumJ (I prefer Kotlin over JS).  Example of tests written in Kotlin via EthereumJ can be found here: https://github.com/Zoltu/recoverable-wallet/blob/master/tests/RecoverableWallet.kt  Truffle has better docs and support network though I believe.

**Comment by branciard** [comment]
I agree with you. its helps for validation, regression and understand and have a clear vision of all usecases.  Even when i audit code, i make truffle tests. It helps me to check basic usecases and then twist on those tests to see robustness or bugs. truffle tests here for instance : https://github.com/branciard/multisigundertests/blob/master/test/wallet.js
